common/scala/src/main/scala/whisk/http/PoolingRestClient.scala

```diff
@@ -55,16 +63,18 @@ class PoolingRestClient(protocol: String, host: String, port: Int, queueSize: In
     Http().cachedHostConnectionPoolHttps[Promise[HttpResponse]](host = host, port = port)
   }
 
+  private val defaultHttpFlow = pool.mapMaterializedValue { x =>
+    poolPromise.success(x); x
+  }
```

Comment:
Have you checked my comments in #3416 towards possible alternatives here? (and the relevant tests)